app/com/linkedin/drelephant/tuning/IPSOManager.java

```diff
@@ -0,0 +1,357 @@
+      tuningParameterConstraint.constraintType = TuningParameterConstraint.ConstraintType.BOUNDARY;
+      tuningParameterConstraint.paramName = tuningParameter.paramName;
+      tuningParameterConstraint.save();
+    }
+  }
+
+  @Override
+  public void extractParameterInformation(List<AppResult> appResults) {
+    logger.info(" Extract Parameter Information");
+    usageDataGlobal = new HashMap<String, Map<String, Double>>();
+    intialize();
+    for (AppResult appResult : appResults) {
+      Map<String, Map<String, Double>> usageDataApplicationlocal = collectUsageDataPerApplication(appResult);
+      for (String functionType : usageDataApplicationlocal.keySet()) {
+        Map<String, Double> usageDataForFunctionGlobal = usageDataGlobal.get(functionType);
+        Map<String, Double> usageDataForFunctionlocal = usageDataApplicationlocal.get(functionType);
+        for (String usageName : usageDataForFunctionlocal.keySet()) {
+          usageDataForFunctionGlobal.put(usageName,
+              max(usageDataForFunctionGlobal.get(usageName), usageDataForFunctionlocal.get(usageName)));
+        }
+      }
+    }
+    logger.info("Usage Values Global ");
+    printInformation(usageDataGlobal);
+  }
+
+  private void intialize() {
+    for (String function : functionTypes) {
+      Map<String, Double> usageData = new HashMap<String, Double>();
+      for (UTILIZED_PARAMETER_KEYS value : UTILIZED_PARAMETER_KEYS.values()) {
+        usageData.put(value.getValue(), 0.0);
+      }
+      usageDataGlobal.put(function, usageData);
+    }
+  }
+
+  private Map<String, Map<String, Double>> collectUsageDataPerApplication(AppResult appResult) {
+    Map<String, Map<String, Double>> usageData = new HashMap<String, Map<String, Double>>();
+    if (appResult.yarnAppHeuristicResults != null) {
+      for (AppHeuristicResult appHeuristicResult : appResult.yarnAppHeuristicResults) {
+        if (appHeuristicResult.heuristicName.equals("Mapper Memory")) {
+          Map<String, Double> counterData = new HashMap<String, Double>();
+          collectUsageDataPerApplicationForFunction(appHeuristicResult, counterData);
+          usageData.put("map", counterData);
+        }
+        if (appHeuristicResult.heuristicName.equals("Reducer Memory")) {
+          Map<String, Double> counterData = new HashMap<String, Double>();
+          collectUsageDataPerApplicationForFunction(appHeuristicResult, counterData);
+          usageData.put("reduce", counterData);
+        }
+      }
+    }
+    logger.info("Usage Values local   " + appResult.jobExecUrl);
+    printInformation(usageData);
+    return usageData;
+  }
+
+  private void printInformation(Map<String, Map<String, Double>> information) {
+    for (String functionType : information.keySet()) {
+      logger.info("function Type    " + functionType);
+      Map<String, Double> usage = information.get(functionType);
+      for (String data : usage.keySet()) {
+        logger.info(data + " " + usage.get(data));
```

Comment:
Check all logger statements and convert them to debug if not a necessary information.